prow/cluster/jobs/istio/test-infra/istio.test-infra.trusted.master.yaml

```diff
@@ -3,7 +3,7 @@ postsubmits:
   istio/test-infra:
   - name: post-test-infra-deploy-prow
     cluster: test-infra-trusted
-    run_if_changed: '^prow/cluster/(?:build/)?[^/]+\.yaml$'
+    run_if_changed: '^prow/cluster/(?:build/|private/|[^/]+\.yaml$)'
```

Comment:
nit: This will run for changes to non-YAML files in the `build` and `private` directories.

Reply:
Should the regexp match the Makefile? Maybe we want this to run when anything in these directories changes regardless of extension?

Reply:
> nit: This will run for changes to non-YAML files in the build and private directories.

The goal is to reserve the `build/` and `public/` as dedicated directories for Kubernetes resource directories. At least that is how we are using those directories currently.

> Should the regexp match the Makefile? 

Because the Makefile has the `gen-private-jobs` target, I am worried that this will trigger unexpected deployments.

> Maybe we want this to run when anything in these directories changes regardless of extension?

The [jobs](https://github.com/istio/test-infra/tree/master/prow/cluster/jobs) are in this directory too; we will want to *exclude* those.